experiments/Atmos_LES/bomex.jl

```diff
@@ -0,0 +1,503 @@
+
+# ---------------------------- Begin Boundary Conditions ----------------- #
+"""
+  BOMEX_BC <: BoundaryCondition
+  Prescribes boundary conditions for Dynamics of Marine Stratocumulus Case
+#Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct BOMEX_BC{FT} <: BoundaryCondition
+  "Friction velocity"
+  u_star::FT  
+  "Sensible Heat Flux"
+  w′θ′::FT
+  "Latent Heat Flux"
+  w′qt′::FT
+end
+
+"""
+    atmos_boundary_state!(nf::Union{NumericalFluxNonDiffusive, NumericalFluxGradient},
+                          bc::BOMEX_BC, args...)
+
+For the non-diffussive and gradient terms we just use the `NoFluxBC`
+"""
+atmos_boundary_state!(nf::Union{NumericalFluxNonDiffusive, NumericalFluxGradient},
+                      bc::BOMEX_BC, 
+                      args...) = atmos_boundary_state!(nf, NoFluxBC(), args...)
+
+"""
+    atmos_boundary_flux_diffusive!(nf::NumericalFluxDiffusive,
+                                   bc::BOMEX_BC, atmos::AtmosModel,
+                                   F,
+                                   state⁺, diff⁺, aux⁺, n⁻,
+                                   state⁻, diff⁻, aux⁻,
+                                   bctype, t,
+                                   state1⁻, diff1⁻, aux1⁻)
+
+When `bctype == 1` the `NoFluxBC` otherwise the specialized BOMEX BC is used
+"""
+#TODO This needs to be in sync with the new boundary condition interfaces
+function atmos_boundary_flux_diffusive!(nf::CentralNumericalFluxDiffusive,
+                                        bc::BOMEX_BC, 
+                                        atmos::AtmosModel, F,
+                                        state⁺, diff⁺, hyperdiff⁺, aux⁺,
+                                        n⁻,
+                                        state⁻, diff⁻, hyperdiff⁻, aux⁻,
+                                        bctype, t,
+                                        state1⁻, diff1⁻, aux1⁻)
+  
+  # Floatint point precision
+  FT = eltype(state⁺)
+
+  # Establish the thermodynamic state based on the prognostic variable state
+  TS = thermo_state(atmos.moisture, atmos.orientation, state⁺, aux⁺)
+
+  # Air temperature given current thermodynamic state
+  temperature = air_temperature(TS)
+
+  # Boundary condition for bottom wall
+  if bctype != 1
+    atmos_boundary_flux_diffusive!(nf, NoFluxBC(), atmos, F,
+                                   state⁺, diff⁺, hyperdiff⁺, aux⁺, n⁻,
+                                   state⁻, diff⁻, hyperdiff⁻, aux⁻,
+                                   bctype, t,
+                                   state1⁻, diff1⁻, aux1⁻)
+  else
+    # Start with the noflux BC and then build custom flux from there
+    atmos_boundary_state!(nf, NoFluxBC(), atmos,
+                          state⁺, diff⁺, aux⁺, n⁻,
+                          state⁻, diff⁻, aux⁻,
+                          bctype, t)
+    # Interior state velocities [u₀]
+    u₀ = state⁻.ρu / state⁻.ρ
```

Comment:
We'd want this to be wind in the interior, i.e., on nodal point inward from the boundary (not right on the boundary). Is it?

Reply:
Yes - notation has been clarified.